blog/2019-10-02-MigratingOptaWebEmployeeRosteringToSpringBoot.adoc

```diff
@@ -0,0 +1,195 @@
+Here's an example of executing a SQL statement that will select every `Skill` entity:
+
+image::skillsConsole.png[]
+
+Here we see the autogenerated demo data saved in the H2 database when the application starts up.
+This console makes it convenient to test and modify the in-memory database.
+
+Ok, so having data generated on startup is nice for demoing the application.
+But what if we wanted to use the application in production and start with no demo data?
+To solve this, we added a feature that gives the user the option to start the application with no persisted data.
+This was implemented by adding an additional system property called `InitialData`:
+
+[source,java]
+----
+public static final String INITIAL_DATA_PROPERTY = "optaweb.generator.initial.data";
+
+public enum InitialData {
+    EMPTY,
+    DEMO_DATA // default
+}
+----
+
+In `RosterGenerator.java`, we use the system property to determine whether to generate data or not.
+Notice that if `initialData = EMPTY`, the method returns without generating any data:
+
+[source,java]
+----
+SystemPropertiesRetriever.InitialData initialData = SystemPropertiesRetriever.determineInitialData();
+
+switch (initialData) {
+    case EMPTY:
+        return;
+    case DEMO_DATA:
+        //Generate demo data
+}
+----
+
+To configure the system properties and start with an empty database, add this to the `applications.properties` file:
+
+[source,properties]
+----
+optaweb.generator.initial.data=EMPTY
+----
+
+This sets the `optaweb.generator.initial.data` system property for the application when it starts.
+
+== REST API
+
+Now let's actually interact with the backend application.
+An easy way to test the REST API is to use the Swagger framework, which OptaWeb Employee Rostering supports.
+
+As an example, here are the `Tenant` REST methods shown in the Swagger user interface:
+
+image::tenantMethods.png[]
+
+As we can see, each method has documentation, the relative path URL, and its corresponding HTTP request.
+The Swagger UI makes it very easy to test any method in the backend.
+Let's say we wanted a list of all the Tenants in the application.
+We start by selecting the corresponding method to display the method details:
+
+image::getTenantsMethod.png[]
+
+Here, the `Example Value` field shows a sample value of a Tenant list, serialized to JSON format.
+We see that each Tenant contains the `id`, `name`, and `version` fields, which are saved in the database whenever a Tenant is added.
+Also note the `Try it out` button in the top right corner.
+We can use it to send HTTP requests with ease for testing purposes.
+Let's try it out:
+
+image::executeGetTenants.png[]
+
+An `Execute` button will appear, and clicking on it sends a GET request that gets all the saved Tenants in JSON format.
+Notice the GET request is sent to the `Request URL`, and the response is found in the `Response body`.
+The server returns the `200` code, which means the request was handled with no issues.
```

Comment:
In my opinion, you are spending too much time explaining how to use the Swagger interface. Can we assume they know? Can we point them to Swagger documentation?

Reply:
The goal here is not so much to show how to use the Swagger interface, but to show the basics of an HTTP request (what the request does, where the request is sent, what the request returns).

I think Swagger shows these basics quite well, which is why I took the readers through the step-by-step process. As I'm writing this, I realize that showing how an HTTP request works is probably not in the scope of this blog. I'll shorten this so we only show the results of the HTTP request.

Showing Swagger docs isn't the goal, so I'm going to assume readers know what an HTTP request is.